yt_dlp/extractor/instagram.py

```diff
@@ -174,6 +186,12 @@ def _extract_product(self, product_info):
         }
         carousel_media = product_info.get('carousel_media')
         if carousel_media:
+            if self.get_param('noplaylist'):
```

Comment:
Adding new features in the middle of review is annoying to deal with. Revert this and you can later open a new PR for this

Reply:
It's a very minor change and would it matter to just add it.

Reply:
Done